base/twiceprecision.jl

```diff
@@ -535,7 +535,22 @@ function __convertSRL(::Type{StepRangeLen{T,R,S}}, r::AbstractRange{U}) where {T
     StepRangeLen{T,R,S}(R(first(r)), S(step(r)), length(r))
 end
 
-function sum(r::StepRangeLen)
+function sum(r::StepRangeLen{T,Float64,Float64}) where {T}
+    l = length(r)
+    # Compute the contribution of step over all indices.
+    # Indexes on opposite side of r.offset contribute with opposite sign,
+    #    r.step * (sum(1:np) - sum(1:nn))
+    np, nn = l - r.offset, r.offset - 1  # positive, negative
+    # To prevent overflow in sum(1:n), multiply its factors by the step
+    sp, sn = sumpair(np), sumpair(nn)
+    tp = r.step * sp[1] * sp[2]
+    tn = r.step * sn[1] * sn[2]
+    s = tp - tn
+    # Add in contributions of ref
+    ref = r.ref * l
+    convert(T, s + ref)
+end
+function sum(r::StepRangeLen{T,Base.TwicePrecision{T},Base.TwicePrecision{T}}) where {T}
```

Comment:
Probably `sum(r::StepRangeLen{T,<:TwicePrecision,<:TwicePrecision}) where {T}` would be a better declaration here.